desktop_version/src/Music.cpp

```diff
@@ -372,14 +372,16 @@ float SoundTrack::volume = 0.0f;
 class MusicTrack
 {
 public:
-    MusicTrack(SDL_RWops *rw)
+    MusicTrack(SDL_RWops *rw, const char* name, bool loose_extra)
     {
         SDL_zerop(this);
         read_buf = (Uint8*) SDL_malloc(rw->size(rw));
         SDL_RWread(rw, read_buf, rw->size(rw), 1);
         int err;
         stb_vorbis_info vorbis_info;
         stb_vorbis_comment vorbis_comment;
+        filename = SDL_strdup(name);
```

Comment:
`name`, `filename` and `id` are used a bit confusingly. In the enumeration code, the file path (with folder and extension) is first written to a buffer `name`, then a plain version without folder or extension is written to a buffer `id`, this is passed to the `MusicTrack` constructor as `name` and then written to the `filename` attribute, which can be requested with a function called `getid`...

Reply:
What do you suggest?

Reply:
Sorry for the late reply!

I'd say:
- `char name[256]` (stores e.g. `music/mysong.ogg`) -> `asset_filename`? It's used as an argument to `FILESYSTEM_loadAssetToMemory()`
- `char id[256]` (stores e.g. `mysong`) -> probably fine as-is
- `char* filename` in `MusicTrack` -> `id` to match
- `const char* name` in the `MusicTrack` constructor -> `id` to match

The names of the constructor argument and class attribute will clash - I'm not too used to C++ OO but I've usually seen the solution in other languages be something like `this.id = id;`. Otherwise, naming the argument `_id` and doing `id = SDL_strdup(_id);` would in my opinion be better than working around it with `filename`/`name`, `loose_extra`/`looseextra`, etc. (And that kind of `_` prefixing is done in other places in the code as well anyway)